### ISSUE TITLE
Bad images in training

### ISSUE THREAD

**Issue by versatran01** [wohlert/generative-query-network-pytorch#11]
While playing around with the sm5 dataset, I noticed some of them are badly rendered.
![individualimage](https://user-images.githubusercontent.com/1883423/46956950-887cfe00-d064-11e8-85c5-f2acfa73132d.png)
Not sure if this will pose any problem for training, just wanted to point this out.

**Comment by wohlert** [comment]
Are you using the dataset you created yourself?

**Comment by versatran01** [comment]
Yes. I've seen several sequences with more than 2 objects in it, like the one I posted here. The image in row 2 column 2 and row 3 column 1 clearly shows 2 different objects (one piece with 3 blocks and another piece with 2 blocks (yellow, blue)).

**Comment by Paandaman** [comment]
I've created my own dataloader based on https://github.com/l3robot/gqn_datasets_translator and I've observed the same kind of poor quality samples. Might such samples cause exploding gradients in the network?

**Comment by versatran01** [comment]
I believe the problem with these images is that the object is not at the center, but the rendering is actually correct.

**Comment by Paandaman** [comment]
Yea I agree, but having some non-centered data(as well as multiple objects in an image) is not representative of the rest of the data on which the network is trying to learn so I would guess it could result in large errors. I'm not sure whether it becomes a problem or not if the amount of such images is small.

**Comment by versatran01** [comment]
I was wondering if you could detect this kind of stuff by looking at the pose? for example if one fit a sphere to the translations of normal ones, the center is at (0, 0, 0) while that of a bad one would be off by a bit.

**Comment by wohlert** [comment]
It looks to only be a few examples at this point, so I conclude it is a problem from the side of DeepMind rather than the converter.
